packages/edit-post/src/store/actions.js

```diff
@@ -492,6 +492,22 @@ export function* __unstableSwitchToTemplateMode( template ) {
 			'wp_template',
 			template
 		);
+
+		const settings = yield controls.select(
+			editorStore.name,
+			'getEditorSettings'
+		);
+
+		const newAvailableTemplates = {
+			...settings.availableTemplates,
+			[ savedTemplate.slug ]: savedTemplate.title.rendered,
+		};
+
+		yield controls.dispatch( editorStore, 'updateEditorSettings', {
+			...settings,
+			availableTemplates: newAvailableTemplates,
+		} );
```

Comment:
While this works, it feels a bit like a hack to me because it address only one use-case: the template being created from the template sidebar. Other plugins could have their own UIs and rely on actions directly.

 I think the ideal solution for me would be to drop the `availableTemplates` entirely and use a `getEntityRecords` selector call to fill the the template selector instead.

Do you think it's possible to explore this alternative and compare solutions?

Reply:
@youknowriad Thanks! I've modified it to use getEntityRecords.

Reply:
I actually think you misunderstand my suggestion 😬. I was proposing to keep this PR as is and open another when we don't update the action but instead update the template selector UI to not rely on `getSettings().availableTemplates` but to use `getEntityRecords` call directly (or an alternative REST API endpoint).

Reply:
@youknowriad 

I understand.

I opened PR https://github.com/WordPress/gutenberg/pull/32744. Please review it.

Thanks!